SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialSpanIdCache.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "Misc/DateTime.h"
+#include "SpatialView/EntityComponentId.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialSpanIdStore, Log, All);
+
+namespace SpatialGDK
+{
+class SpatialSpanIdCache
+{
+public:
+	struct FieldSpanIdUpdate
+	{
+		uint32 FieldId;
+		worker::c::Trace_SpanId NewSpanId;
+		worker::c::Trace_SpanId OldSpanId;
+	};
+
+	void ComponentAdd(const Worker_Op& Op);
+	bool ComponentRemove(const Worker_Op& Op);
+	TArray<FieldSpanIdUpdate> ComponentUpdate(const Worker_Op& Op);
+
+	void AddSpanId(const EntityComponentId& Id, const uint32 FieldId, worker::c::Trace_SpanId SpanId);
+	bool DropSpanId(const EntityComponentId& Id, const uint32 FieldId);
+	bool DropSpanIds(const EntityComponentId& Id);
+
+	bool GetSpanId(const EntityComponentId& Id, const uint32 FieldId, worker::c::Trace_SpanId& OutSpanId, bool bRemove = true);
+	bool GetMostRecentSpanId(const EntityComponentId& Id, worker::c::Trace_SpanId& OutSpanId, bool bRemove = true);
+
+private:
+	struct EntityComponentFieldId
+	{
+		EntityComponentId EntityComponentId;
+		uint32 FieldId;
+	};
+
+	struct EntityComponentFieldIdSpanIdUpdate
+	{
+		worker::c::Trace_SpanId SpanId = worker::c::Trace_SpanId();
+		FDateTime UpdateTime;
+	};
+
+	int32 CompactCounter;
+	int32 CompactFrequency = 100;
+
+	using FieldIdMap = TMap<uint32, EntityComponentFieldIdSpanIdUpdate>;
```

Comment:
Idle thought - the keys for this map are roughly the integers from 1 to total number of fields. Might be worth considering direct mapping rather than a hash map.